woocommerce/google-listings-and-ads
# Refactor product issues

### Changes proposed in this Pull Request:



Part of https://github.com/woocommerce/google-listings-and-ads/issues/2146 & https://github.com/woocommerce/google-listings-and-ads/issues/2250

This PR fetches the product issues using the same background job used to fetch the product statuses. This allows us to reuse the WC products that were fetched while updating the product statuses.

The logic remains mostly unchanged from before. The main difference is that previously, Google IDs were stored in the database and used to fetch the issues. With the new sync mechanism, since we don't have the Google IDs until we fetch all product statuses, we now use the data from the product statuses to obtain the Google IDs and fetch the issues.

### Screenshots:




### Detailed test instructions:


1. Set your PHP memory limit low so you can catch any heavy process. For example, setting this up to 128MB.
2. Call `GET wp-json/wc/gla/mc/product-statistics/refresh`
3. Go to WooCommerce -> Status -> Action Scheduler -> Search for `gla/jobs/update_merchant_product_statuses/process_item` see how the jobs get completed successfully.
4. See how the table `wp_gla_merchant_issues` gets populated.
5. Once all jobs are completed call `GET wp-json/wc/gla/mc/product-statistics` and see that you get the right stats.


### Additional details:
 - Account and pre-sync issues are not migrated yet.
 - After all the issues logic is refactored, I will remove any legacy code.
 - I wanted to address the scenario where the job fails and some issues are inserted. I used the "created at" field, but it appears that the field will be identical for all batches. Therefore, I'll need to see alternative solutions in a follow-up PR. Any suggestions are welcome too.
 

### Changelog entry

>
